### PR TITLE
Add and test `empty` method

### PR DESCRIPTION
**Pull request by amsross**
`empty` creates a stream with no values.

This turns a Highland stream into a [Monoid](https://github.com/fantasyland/fantasy-land#monoid)

### CODE REVIEW

**Comment by svozza** [comment]
Don't Fantasyland method names need to be prefixed now? E.g., `fantasyland/empty`.

**Comment by amsross** [comment]
@svozza fantasyland prefixes are being added in #658. Since those don't currently exist in `master`, I'd prefer to add the functionality in this PR, merge to `master`, rebase #658 against those changes, and add the appropriate prefixes in that PR.

This also applies to #643.

File: lib/index.js [hunk 1]
```diff
@@ -1001,6 +1001,22 @@ addToplevelMethod('of', function (x) {
...
+ * @section Utils
+ * @name _.empty()
+ * @returns Stream
+ * @api public
+ *
+ * _.empty().toArray(_.log); // => []
+ */
+
+addToplevelMethod('empty', function () {
+    return _([]);
```
Comment (vqvu) [inline lib/index.js]: Use `this([])` instead of `_([])`.

There's a new feature in 3.0 (not yet documented) called `use` that lets you extend the stream prototype. Using `this` will make sure that you construct the new stream using the correct prototype.

Note that `of` has the same bug.

**Comment by vqvu** [comment]
Looks good!
